### PR TITLE
[PXP-2658] Upgrade PyYAML to 5.1 and use safe_load()

### PR DESCRIPTION
**Pull request by vpsx**
Description about what this pull request does.

Please make sure to follow the [DEV guidelines](https://gen3.org/resources/developer/dev-introduction/) before asking for review.

### Improvements
Use PyYaml safe_load() instead of load()

### Dependency updates
Upgrade PyYaml to 5.1

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
The style in this PR agrees with [`black`](https://github.com/ambv/black). :heavy_check_mark:

This formatting comment was generated automatically by a script in [uc-cdis/wool](https://github.com/uc-cdis/wool).

**Comment by PlanXCyborg** [comment]
Jenkins Build 1 : time taken 1 hr 20 min
Check the https://jenkins.planx-pla.net/job/perf-sheepdog/job/PR-271/1/display/redirect


Test results:
	Passed: 109, Failed: 0, Skipped: 0



| Test | Time (`PR`) | Time (`master`) | Diff |
| - | - | - | - |
| @ExportPerformanceTests @Performance: Export all nodes of type: read_group | 0.634 | 0.358 | 0.28 |
| @ExportPerformanceTests @Performance: Export all nodes of type: read_group_qc | 0.102 | 0.066 | 0.04 |
| @ExportPerformanceTests @Performance: Export all nodes of type: sample | 0.441 | 0.075 | 0.37 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: aliquot | 0.165 | 0.07 | 0.10 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: case | 0.269 | 0.065 | 0.20 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: sample | 0.184 | 0.112 | 0.07 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 0 | 0.578 | 0.426 | 0.15 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 1 | 0.552 | 0.169 | 0.38 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 2 | 0.541 | 0.393 | 0.15 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 3 | 0.429 | 0.24 | 0.19 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 5 | 0.262 | 0.236 | 0.03 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 7 | 0.079 | 0.063 | 0.02 |
| @QueryPerformanceTests @Performance: Executing topDown query # 1 | 0.084 | 0.061 | 0.02 |
| @QueryPerformanceTests @Performance: Executing topDown query # 2 | 0.103 | 0.066 | 0.04 |
| @QueryPerformanceTests @Performance: Executing topDown query # 3 | 0.171 | 0.109 | 0.06 |
| @QueryPerformanceTests @Performance: Executing topDown query # 4 | 0.269 | 0.112 | 0.16 |
| @QueryPerformanceTests @Performance: Executing topDown query # 7 | 0.533 | 0.472 | 0.06 |
| @SubmissionPerformanceTests @Performance: Submission acknowledgement 100 | 2.502 | 0.417 | 2.09 |
| @SubmissionPerformanceTests @Performance: Submission acknowledgement 1000 | 23.528 | 0.291 | 23.24 |
| @SubmissionPerformanceTests @Performance: Submission aligned_reads_index 10 | 1.852 | 0.221 | 1.63 |
| @SubmissionPerformanceTests @Performance: Submission aligned_reads_index 100 | 16.745 | 0.138 | 16.61 |
| @SubmissionPerformanceTests @Performance: Submission aligned_reads_index 1000 | 45.014 | 0.075 | 44.94 |
| @SubmissionPerformanceTests @Performance: Submission aliquot 10 | 0.378 | 0.065 | 0.31 |
| @SubmissionPerformanceTests @Performance: Submission aliquot 100 | 2.435 | 0.081 | 2.35 |
| @SubmissionPerformanceTests @Performance: Submission aliquot 1000 | 22.863 | 0.082 | 22.78 |
| @SubmissionPerformanceTests @Performance: Submission case 10 | 0.43 | 0.167 | 0.26 |
| @SubmissionPerformanceTests @Performance: Submission case 100 | 2.323 | 0.235 | 2.09 |
| @SubmissionPerformanceTests @Performance: Submission case 1000 | 24.335 | 0.36 | 23.97 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 10 | 0.57 | 0.512 | 0.06 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 100 | 2.775 | 0.501 | 2.27 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 1000 | 26.755 | 0.304 | 26.45 |
| @SubmissionPerformanceTests @Performance: Submission core_metadata_collection 1000 | 24.357 | 2.041 | 22.32 |
| @SubmissionPerformanceTests @Performance: Submission demographic 1000 | 25.204 | 0.295 | 24.91 |
| @SubmissionPerformanceTests @Performance: Submission diagnosis 1000 | 28.336 | 0.438 | 27.90 |
| @SubmissionPerformanceTests @Performance: Submission experiment 1000 | 23.918 | 0.381 | 23.54 |
| @SubmissionPerformanceTests @Performance: Submission experimental_metadata 1000 | 45.013 | 0.918 | 44.10 |
| @SubmissionPerformanceTests @Performance: Submission exposure 1000 | 23.678 | 0.326 | 23.35 |
| @SubmissionPerformanceTests @Performance: Submission family_history 1000 | 23.891 | 0.36 | 23.53 |
| @SubmissionPerformanceTests @Performance: Submission keyword 1000 | 22.097 | 0.348 | 21.75 |
| @SubmissionPerformanceTests @Performance: Submission publication 1000 | 21.836 | 2.088 | 19.75 |
| @SubmissionPerformanceTests @Performance: Submission read_group 1000 | 26.76 | 0.338 | 26.42 |
| @SubmissionPerformanceTests @Performance: Submission read_group_qc 1000 | 23.394 | 0.336 | 23.06 |
| @SubmissionPerformanceTests @Performance: Submission sample 1000 | 27.927 | 0.325 | 27.60 |
| @SubmissionPerformanceTests @Performance: Submission slide 1000 | 26.151 | 0.354 | 25.80 |
| @SubmissionPerformanceTests @Performance: Submission slide_count 1000 | 25.058 | 0.313 | 24.75 |
| @SubmissionPerformanceTests @Performance: Submission slide_image 1000 | 44.667 | 0.482 | 44.18 |
| @SubmissionPerformanceTests @Performance: Submission submitted_aligned_reads 1000 | 44.253 | 0.538 | 43.71 |
| @SubmissionPerformanceTests @Performance: Submission submitted_copy_number 1000 | 45.202 | 0.409 | 44.79 |
| @SubmissionPerformanceTests @Performance: Submission submitted_methylation 10 | 2.748 | 1.985 | 0.76 |
| @SubmissionPerformanceTests @Performance: Submission submitted_methylation 1000 | 44.543 | 0.324 | 44.22 |
| @SubmissionPerformanceTests @Performance: Submission submitted_somatic_mutation 1000 | 45.014 | 2.016 | 43.00 |
| @SubmissionPerformanceTests @Performance: Submission submitted_unaligned_reads 1000 | 44.513 | 1.967 | 42.55 |
| @SubmissionPerformanceTests @Performance: Submission treatment 1000 | 22.948 | 2.053 | 20.89 |

File: dev-requirements.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@ codacy-coverage
 Sphinx==1.6.5
 sphinx_rtd_theme
 flasgger==0.9.1
-PyYAML==4.2b1
```
Comment (fantix) [inline dev-requirements.txt]: (trivial) as Pauline mentioned in a different PR that I cannot remember which, maybe it is okay to remove this duplicate requirement from `dev-requirements.txt` if it already exists in `requirements.txt`?

Comment (vpsx) [reply dev-requirements.txt]: O yes indeed thank you 🤦‍♂️

**Comment by PlanXCyborg** [comment]
Jenkins Build 4 : time taken 40 min
Check the https://jenkins.planx-pla.net/job/perf-sheepdog/job/PR-271/4/display/redirect


Test results:
	Passed: 109, Failed: 0, Skipped: 0



| Test | Time (`PR`) | Time (`master`) | Diff |
| - | - | - | - |
| @ExportPerformanceTests @Performance: Export all nodes of type: experiment | 0.799 | 0.374 | 0.43 |
| @ExportPerformanceTests @Performance: Export all nodes of type: program | 0.104 | 0.078 | 0.03 |
| @ExportPerformanceTests @Performance: Export all nodes of type: project | 0.341 | 0.112 | 0.23 |
| @ExportPerformanceTests @Performance: Export all nodes of type: sample | 0.667 | 0.443 | 0.22 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: aliquot | 0.271 | 0.236 | 0.03 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: case | 0.367 | 0.28 | 0.09 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: experiment | 0.266 | 0.232 | 0.03 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: read_group | 0.087 | 0.07 | 0.02 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: sample | 0.137 | 0.111 | 0.03 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 3 | 0.575 | 0.507 | 0.07 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 4 | 0.494 | 0.443 | 0.05 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 5 | 0.386 | 0.291 | 0.09 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 6 | 0.271 | 0.159 | 0.11 |
| @QueryPerformanceTests @Performance: Executing topDown query # 0 | 0.093 | 0.06 | 0.03 |
| @QueryPerformanceTests @Performance: Executing topDown query # 1 | 0.141 | 0.076 | 0.07 |
| @QueryPerformanceTests @Performance: Executing topDown query # 2 | 0.127 | 0.118 | 0.01 |
| @QueryPerformanceTests @Performance: Executing topDown query # 7 | 0.78 | 0.59 | 0.19 |
| @SubmissionPerformanceTests @Performance: Submission aligned_reads_index 10 | 2.248 | 2.044 | 0.20 |
| @SubmissionPerformanceTests @Performance: Submission aligned_reads_index 100 | 20.904 | 17.882 | 3.02 |
| @SubmissionPerformanceTests @Performance: Submission case 100 | 2.884 | 2.481 | 0.40 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 100 | 3.385 | 2.978 | 0.41 |
| @SubmissionPerformanceTests @Performance: Submission core_metadata_collection 100 | 4.098 | 2.879 | 1.22 |
| @SubmissionPerformanceTests @Performance: Submission experimental_metadata 10 | 2.556 | 1.897 | 0.66 |
| @SubmissionPerformanceTests @Performance: Submission experimental_metadata 100 | 21.427 | 16.769 | 4.66 |
| @SubmissionPerformanceTests @Performance: Submission publication 10 | 0.397 | 0.375 | 0.02 |
| @SubmissionPerformanceTests @Performance: Submission read_group 100 | 2.898 | 2.608 | 0.29 |
| @SubmissionPerformanceTests @Performance: Submission read_group_qc 100 | 4.533 | 3.891 | 0.64 |
| @SubmissionPerformanceTests @Performance: Submission sample 100 | 3.188 | 2.946 | 0.24 |
| @SubmissionPerformanceTests @Performance: Submission slide 10 | 0.472 | 0.394 | 0.08 |
| @SubmissionPerformanceTests @Performance: Submission slide_count 10 | 0.586 | 0.38 | 0.21 |
| @SubmissionPerformanceTests @Performance: Submission slide_image 100 | 22.329 | 17.28 | 5.05 |
| @SubmissionPerformanceTests @Performance: Submission submitted_aligned_reads 10 | 2.244 | 1.948 | 0.30 |
| @SubmissionPerformanceTests @Performance: Submission submitted_aligned_reads 100 | 21.223 | 17.434 | 3.79 |
| @SubmissionPerformanceTests @Performance: Submission submitted_copy_number 10 | 2.124 | 1.819 | 0.31 |
| @SubmissionPerformanceTests @Performance: Submission submitted_copy_number 100 | 21.398 | 16.438 | 4.96 |
| @SubmissionPerformanceTests @Performance: Submission submitted_methylation 100 | 21.457 | 17.376 | 4.08 |
| @SubmissionPerformanceTests @Performance: Submission submitted_somatic_mutation 10 | 2.316 | 2.104 | 0.21 |
| @SubmissionPerformanceTests @Performance: Submission submitted_somatic_mutation 100 | 22.784 | 17.331 | 5.45 |
| @SubmissionPerformanceTests @Performance: Submission submitted_unaligned_reads 10 | 2.476 | 2.103 | 0.37 |
| @SubmissionPerformanceTests @Performance: Submission submitted_unaligned_reads 100 | 21.262 | 17.484 | 3.78 |

**Pull request merged**